addons/stock/models/product.py

```diff
@@ -7,6 +7,7 @@
 from dateutil.relativedelta import relativedelta
 
 from odoo import _, api, fields, models
+from odoo.addons.barcodes.models.barcode_nomenclature import BarcodeNomenclature
```

Comment:
These changes should be in `product`, not in `stock`, otherwise the issue can happen with modules who depend of `product` but not of `stock` (`purchase` or `sale_management` for example).
Maybe for `product.product` and `product.template`, you can copy-paste the code... I'm not a fan of code duplication but except placing the method directly in `web` (which could makes sense I think since the route to print barcodes is in `web` too, but I really don't know where to place it ?), I don't see other good solution :/
An opinion about that @Feyensv ?

Reply:
IMHO, if the check doesn't require anything from the `barcodes` module itself, it could be a tooling in `odoo.tools` directly (doesn't make any sense to put that in `web`).

Also, since I'm only noticing now that this is yet another 'dumb user' sentry PR, I doubt we really need to fix this in earlier versions, master should be enough.